File: app/src/main/java/com/bkjcb/rqapplication/fragment/BaseFragment.java
package com.bkjcb.rqapplication.fragment;

import android.support.v4.app.Fragment;

import io.reactivex.disposables.Disposable;

/**
 * Created by DengShuai on 2019/2/19.
 * Description :
 */
public class BaseFragment extends Fragment {
    protected Disposable disposable;
    @Override
    public void onStop() {
        super.onStop();

    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (disposable != null && !disposable.isDisposed()) {
            disposable.dispose();
        }
    }
}

File: app/src/main/java/com/bkjcb/rqapplication/base/fragment/BaseFragment.kt
package com.bkjcb.rqapplication.base.fragment

import android.support.v4.app.Fragment
import io.reactivex.disposables.Disposable

/**
 * Created by DengShuai on 2019/2/19.
 * Description :
 */
open class BaseFragment : Fragment() {
    @JvmField
    protected var disposable: Disposable? = null
    override fun onStop() {
        super.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
        if (disposable != null && !disposable!!.isDisposed) {
            disposable?.dispose()
        }
    }
}